### DISCUSSION TITLE
User using AWs IAM without having full root access

### DISCUSSION THREAD

**Discussion by gmichel-paaneah** [pulumi/pulumi#7962] [Pulumi in Practice] [answered]
We have a use case where we would like to delegate AWS account creation and deletion to a third party but we are hesitant to give them root access to the organization or the account. Ideally, can we use Pulumi to automate the process and to use CyberArk or Hashi Vault or Pulumi if you can handle secrets to authenticate to AWS without storing or knowing root access token.

**Comment by leezen** [comment, marked as answer]
This is entirely possible and there are a variety of ways to achieve what you've outlined above in a secure way. One way would be to create [`aws.iam.User`](https://www.pulumi.com/docs/reference/pkg/aws/iam/user/) resources along with corresponding [`aws.iam.AccessKey`](https://www.pulumi.com/docs/reference/pkg/aws/iam/accesskey/) resources. I believe the secrets should be encrypted in state, but you can always apply [`additionalSecretOutputs`](https://www.pulumi.com/docs/intro/concepts/resources/#additionalsecretoutputs) option to ensure this.
